proxy.go

```diff
@@ -358,6 +362,29 @@ func (rp *reverseProxy) serveFromCache(s *scope, srw *statResponseWriter, req *h
 	}
 }
 
+func calcQueryParamsHash(origParams url.Values) uint32 {
```

Comment:
Hi, can you add unit tests on this function?

Reply:
I exported the Hashing functionality to `calcMapHash` and added unit tests on it.
Due to this change, Do you think I should add unit tests on this function too?

Reply:
I'm a bit picky but yes. calcMapHash is doing the hashing part. But the calcQueryParamsHash is doing more than just hashing: it ensures that only the param_ are hashed and if there are no param_ then it returns 0. This logic should be tested.